src/zenml/materializers/numpy_materializer.py

```diff
@@ -145,4 +144,53 @@ def extract_metadata(
             "min": min_val,
             "max": max_val,
         }
-        return {**base_metadata, **numpy_metadata}
+        return numpy_metadata
+
+    def extract_text_metadata(
+        self, arr: "NDArray[Any]"
+    ) -> Dict[str, "MetadataType"]:
+        """Extracts text metadata from a numpy array.
+
+        Args:
+            arr: The numpy array to extract metadata from.
+
+        Returns:
+            A dictionary of metadata.
+        """
+        text = " ".join(arr)
+        words = text.split()
+        word_counts = Counter(words)
+        unique_words = len(word_counts)
+        total_words = len(words)
+        most_common_word, most_common_count = word_counts.most_common(1)[0]
+
+        text_metadata: Dict[str, "MetadataType"] = {
+            "unique_words": unique_words,
+            "total_words": total_words,
+            "most_common_word": most_common_word,
+            "most_common_count": most_common_count,
+        }
+        return text_metadata
+
+    def extract_metadata(
+        self, arr: "NDArray[Any]"
+    ) -> Dict[str, "MetadataType"]:
+        """Extract metadata from the given numpy array.
+
+        Args:
+            arr: The numpy array to extract metadata from.
+
+        Returns:
+            The extracted metadata as a dictionary.
+        """
+        base_metadata = super().extract_metadata(arr)
+        if np.issubdtype(arr.dtype, np.number):
+            return {**base_metadata,**self.extract_numeric_metadata(arr)}
+        elif np.issubdtype(arr.dtype, np.unicode_) or np.issubdtype(
+            arr.dtype, np.object_):
+            return {**base_metadata,**self.extract_text_metadata(arr)}
+        else:
+            logger.info(
```

Comment:
Would suggest to change this to debug or take it out since I think users will have no clue what this message could mean